app/models/concerns/indexable.rb

```diff
@@ -279,19 +279,34 @@ def count
       Elasticsearch::Model.client.count(index: index_name)['count']
     end
 
+    def reindex(index: nil)
+      return nil unless index.present?
+      
+      self.__elasticsearch__.create_index! index: index unless self.__elasticsearch__.index_exists? index: index
+      index_name = self.index_name
+
+      client = Elasticsearch::Model.client
+      ### always take origing index even if it was a alias
+      index_name = client.indices.get_alias(name: index_name).first.first  if client.indices.exists_alias? name: index_name
+      client.reindex body: { source: { index: index_name }, dest: { index: index } }
+      puts "Index #{index} exists? #{self.__elasticsearch__.index_exists? index: index}"
+    end
+
     def create_alias(index: nil)
       return nil unless index.present?
 
-      client     = self.gateway.client
+      client     = Elasticsearch::Model.client
```

Comment:
This should be ok, but have you checked whether this works in AWS?

Reply:
I have not deployed to Stage yet, so I need to see there. Locally its been working well.